tests/test_sca.py

```diff
@@ -275,12 +295,7 @@ def test_slow_software_composition_analysis(
             )
         )
     assert mock_syft.call_args_list == expected_syft_call_arg_list
-    assert Component.objects.filter(purl=expected_purl).exists()
-    if is_container:
-        root_component = Component.objects.get(
-            type=Component.Type.CONTAINER_IMAGE, arch="noarch", software_build=sb
-        )
-    else:
-        root_component = Component.objects.srpms().get(software_build=sb)
-        root_component.get_provides_purls()
+    expected_component = Component.objects.get(purl=expected_purl)
+    assert expected_component
+    assert expected_component.meta_attr["source"] == "syft-0.60.1"
```

Comment:
I think the old version was a more useful test, since it asserted that a component with the right type (either a container image or an SRPM) got created. In the new code we assert the component exists, but don't check it has the right type.

I think you should just be able to ```assert root_component.meta_attr["source"] == "syft-0.60.1"``` after the if / else block, since root_component will be defined either way. There's also no need to ```assert expected_component``` since the .get() call should raise a Component.DoesNotExist exception if it's missing.

Reply:
Ah, it looked weird that we were using `get` to test, but you are right it was useful because if it doesn't exist it will thrown an DoesNotExist error.